packages/web-components/@sfx/base/test/base.test.ts

```diff
@@ -0,0 +1,94 @@
+import { expect, spy, stub } from './utils';
+import Base from '../src/base';
+import { LitElement } from 'lit-element';
+import { TemplateResult } from 'lit-element';
+import * as BaseUtils from '../src/utils';
+
+
+describe('Base Class', () => {
+  let base: any = {}
+  beforeEach(() => {
+    base = new Base();
+  });
+
+  it('should extend from LitElement', () => {
+    const base = new Base();
+    expect(base).to.be.an.instanceof(LitElement);
+  });
+
+  describe('constructor', () => {
+    it('should call addSlots', () => {
+      const addSlotsStub = stub(Base.prototype, 'addSlots')
+      base = new Base();
+      expect(addSlotsStub).to.have.been.called;
+    });
+  });
+
+  describe('first update', () => {
+    it('should disconnect the observer', () => {
+      const disconnect = spy()
+      base.observer = { disconnect }
+      base.firstUpdate()
+      expect(disconnect).to.have.been.called;
+    });
+  });
+
+  describe('connectedCallback', () => {
+    it('should call its super connectedCallback', () => {
+      const litElementConnectedCallbackStub = stub(LitElement.prototype, 'connectedCallback')
+      base.connectedCallback();
+      expect(litElementConnectedCallbackStub).to.have.been.called;
+    });
+
+    it('should call createChildrenObserver on the observer property', () => {
+      const observer = {
```

Comment:
You may need to type this as `any` when you change the type in the class to `MutationObserver`.